planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/start_planner_module.cpp

```diff
@@ -825,21 +825,56 @@ BehaviorModuleOutput StartPlannerModule::plan()
       status_.pull_out_path.end_pose.position);
     updateRTCStatus(start_distance, finish_distance);
+
+    const auto start_idx = autoware::motion_utils::findNearestIndex(
+      path.points, status_.pull_out_path.start_pose.position);
+    const auto finish_idx = autoware::motion_utils::findNearestIndex(
+      path.points, status_.pull_out_path.end_pose.position);
+    const double start_velocity = path.points.at(start_idx).point.longitudinal_velocity_mps;
+    const double finish_velocity = path.points.at(finish_idx).point.longitudinal_velocity_mps;
+
+    const auto pull_out_lanes = start_planner_utils::getPullOutLanes(
+      planner_data_,
+      planner_data_->parameters.backward_path_length + parameters_->max_back_distance);
+    const double start_shift_length =
+      lanelet::utils::getArcCoordinates(pull_out_lanes, status_.pull_out_path.start_pose).distance;
+    const double finish_shift_length =
+      lanelet::utils::getArcCoordinates(pull_out_lanes, status_.pull_out_path.end_pose).distance;
+
     planning_factor_interface_->add(
       start_distance, finish_distance, status_.pull_out_path.start_pose,
       status_.pull_out_path.end_pose, planning_factor_direction,
-      utils::path_safety_checker::to_safety_factor_array(debug_data_.collision_check));
+      utils::path_safety_checker::to_safety_factor_array(debug_data_.collision_check),
+      status_.driving_forward, start_velocity, finish_velocity, start_shift_length,
+      finish_shift_length);
     setDebugData();
     return output;
   }
   const double distance = autoware::motion_utils::calcSignedArcLength(
     path.points, planner_data_->self_odometry->pose.pose.position,
     status_.pull_out_path.start_pose.position);
   updateRTCStatus(0.0, distance);
+
+  const auto start_idx = autoware::motion_utils::findNearestIndex(
+    path.points, status_.pull_out_path.start_pose.position);
+  const auto finish_idx =
+    autoware::motion_utils::findNearestIndex(path.points, status_.pull_out_path.end_pose.position);
+  const double start_velocity = path.points.at(start_idx).point.longitudinal_velocity_mps;
+  const double finish_velocity = path.points.at(finish_idx).point.longitudinal_velocity_mps;
+
+  const auto pull_out_lanes = start_planner_utils::getPullOutLanes(
+    planner_data_, planner_data_->parameters.backward_path_length + parameters_->max_back_distance);
+  const double start_shift_length =
+    lanelet::utils::getArcCoordinates(pull_out_lanes, status_.pull_out_path.start_pose).distance;
+  const double finish_shift_length =
+    lanelet::utils::getArcCoordinates(pull_out_lanes, status_.pull_out_path.end_pose).distance;
+
   planning_factor_interface_->add(
     0.0, distance, status_.pull_out_path.start_pose, status_.pull_out_path.end_pose,
     planning_factor_direction,
-    utils::path_safety_checker::to_safety_factor_array(debug_data_.collision_check));
+    utils::path_safety_checker::to_safety_factor_array(debug_data_.collision_check),
```

Comment:
@Kotakku 
For control points during backward driving,
shall we set the control point at the position where it switches from backward driving to forward driving?